NETCORE/src/Shared/Extensions/ApplicationInsightsServiceOptions.cs

```diff
@@ -27,14 +31,24 @@ public class ApplicationInsightsServiceOptions
         /// <summary>
         /// Gets or sets a value indicating whether AppServicesHeartbeatTelemetryModule should be enabled.
         /// Defaults to <value>true</value>.
+        /// IMPORTANT: This setting will be disabled if <see cref="EnableDiagnosticsTelemetryModule"/> is set to false.
         /// </summary>
-        public bool EnableAppServicesHeartbeatTelemetryModule { get; set; } = true;
+        public bool EnableAppServicesHeartbeatTelemetryModule
+        {
+            get => this.enableAppServicesHeartbeatTelemetryModule && this.EnableDiagnosticsTelemetryModule;
```

Comment:
Don't think we need to couple this setting with other. i.e
If a customer disables DiagnosticModule, hearbeats won't flow anyway. But `EnableAppServicesHeartbeatTelemetryModule` is still true.

Reply:
If one user adds 
```
service.Add(DiagnosticModule)
service.ConfigureTelemetryModule(diagmodule.Enableheartbeat = false)
```

The above also results in no heartbeats. I think there is no need to link the Setter/Getter fields. They can be independent. The comment should still say the net effect which is - if diagmodule is disabled, then heartbeats are also not sent, including appservice/azure specific heartbeats.